lib/annotations.go

```diff
@@ -0,0 +1,105 @@
+	// creating a checkpoint image to specify the version of the container engine.
+	CheckpointAnnotationEngineVersion = "org.criu.checkpoint.engine.version"
+
+	// CheckpointAnnotationName is used by Container Checkpoint when
+	// creating a checkpoint image to specify the name of the checkpoint.
+	CheckpointAnnotationName = "org.criu.checkpoint.name"
+
+	// CheckpointAnnotationPod is used by Container Checkpoint when
+	// creating a checkpoint image to specify the name of the pod associated
+	// with the checkpoint.
+	CheckpointAnnotationPod = "org.criu.checkpoint.pod.name"
+
+	// CheckpointAnnotationPodID is used by Container Checkpoint when
+	// creating a checkpoint image to specify the ID of the pod associated
+	// with the checkpoint.
+	CheckpointAnnotationPodID = "org.criu.checkpoint.pod.id"
+
+	// CheckpointAnnotationNamespace is used by Container Checkpoint when
+	// creating a checkpoint image to specify the namespace of the pod associated
+	// with the checkpoint.
+	CheckpointAnnotationNamespace = "org.criu.checkpoint.pod.namespace"
+
+	// CheckpointAnnotationRootfsImage is used by Container Checkpoint when
+	// creating a checkpoint image to specify the root filesystem image associated
+	// with the checkpoint.
+	CheckpointAnnotationRootfsImage = "org.criu.checkpoint.rootfsImage"
+
+	// CheckpointAnnotationRootfsImageID is used by Container Checkpoint when
+	// creating a checkpoint image to specify the ID of the root filesystem image
+	// associated with the checkpoint.
+	CheckpointAnnotationRootfsImageID = "org.criu.checkpoint.rootfsImageID"
+
+	// CheckpointAnnotationRootfsImageName is used by Container Checkpoint when
+	// creating a checkpoint image to specify the name of the root filesystem image
+	// associated with the checkpoint.
+	CheckpointAnnotationRootfsImageName = "org.criu.checkpoint.rootfsImageName"
+
+	// CheckpointAnnotationRuntimeName is used by Container Checkpoint when
+	// creating a checkpoint image to specify the runtime used on the host where
+	// the checkpoint was created.
+	CheckpointAnnotationRuntimeName = "org.criu.checkpoint.runtime.name"
+
+	// CheckpointAnnotationRuntimeVersion is used by Container Checkpoint when
+	// creating a checkpoint image to specify the version of runtime used on the
+	// host where the checkpoint was created.
+	CheckpointAnnotationRuntimeVersion = "org.criu.checkpoint.runtime.version"
+
+	// CheckpointAnnotationRawImageName is used by Container Checkpoint when
+	// creating a checkpoint image to specify the original unprocessed name of
+	// the image used to create the container (as specified by the user).
+	CheckpointAnnotationRawImageName = "org.criu.checkpoint.rawImageName"
+
+	// CheckpointAnnotationPodmanVersion is used by Container Checkpoint when
+	// creating a checkpoint image to specify the version of Podman used on the
+	// host where the checkpoint was created.
+	CheckpointAnnotationPodmanVersion = "org.criu.checkpoint.podman.version"
```

Comment:
This is not necessary because we have "org.criu.checkpoint.engine.version"

Reply:
I was copying this type of annotation from the [podman](https://github.com/containers/podman/blob/447d3e2a44922b83495a331528bbd8922277e067/libpod/define/annotations.go#L67) and [CRI-O](https://github.com/cri-o/cri-o/blob/183f28b7e4ec06b004de4d54b771960e83d2f9fb/pkg/annotations/checkpoint.go#L3) as those might be used for creating checkpoint image in that particular engine.

Reply:
As mentioned above, I do not think we need any container engine specific annotations.